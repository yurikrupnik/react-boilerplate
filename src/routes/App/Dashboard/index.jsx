import Loadable from '../../../components/Loadable';

const Mobile = Loadable({
    loader: () => import(/* webpackChunkName: "dashboardMobile" */ './Mobile'),
});


const Desktop = Loadable({
    loader: () => import(/* webpackChunkName: "dashboardDesktop" */ './Desktop'),
});

export default [Mobile, Desktop];
