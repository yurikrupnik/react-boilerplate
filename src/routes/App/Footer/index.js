// import Mobile from './Mobile';
// import Desktop from './Desktop';
import Loadable from '../../../components/Loadable';

const Mobile = Loadable({
    loader: () => import(/* webpackChunkName: "footerMobile" */ './Mobile'),
});


const Desktop = Loadable({
    loader: () => import(/* webpackChunkName: "footerDesk" */ './Desktop'),
});

export default [Mobile, Desktop];
