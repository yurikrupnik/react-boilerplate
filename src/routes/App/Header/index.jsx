import Loadable from '../../../components/Loadable/index';

const Mobile = Loadable({
    loader: () => import(/* webpackChunkName: "headerMobile" */ './Mobile'),
});


const Desktop = Loadable({
    loader: () => import(/* webpackChunkName: "headerDesktop" */ './Desktop'),
});

export default [Mobile, Desktop];
