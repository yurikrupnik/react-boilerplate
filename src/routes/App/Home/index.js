// import Mobile from './Mobile';
// import Desktop from './Desktop';
import Loadable from '../../../components/Loadable/index';

const Mobile = Loadable({
    loader: () => import(/* webpackChunkName: "homeMobile" */ './Mobile'),
});


const Desktop = Loadable({
    loader: () => import(/* webpackChunkName: "homeDesktop" */ './Desktop'),
});

export default [Mobile, Desktop];
