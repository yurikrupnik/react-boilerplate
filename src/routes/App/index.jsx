import Loadable from '../../components/Loadable';

const App = Loadable({
    loader: () => import('./app'),
});

export default App;
