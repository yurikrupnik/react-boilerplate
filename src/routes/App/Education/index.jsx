import React, { Fragment } from 'react';
// import PropTypes from 'prop-types';
import { Route } from 'react-router-dom';
import { Consumer as Device } from '../../services/context/device';
import routes from './routes';

const Education = () => (
    <Device render={(device) => {
        const isMobile = device.isMobile();
        return (
            <Fragment>
                {routes.getRoutesByType(isMobile).map(route => (
                    <Route
                        key={route.key}
                        path={route.path}
                        exact={route.exact}
                        component={route.component}
                    />
                ))}
                <div className="row">
                    <div className="col-xs-3">
                        {routes.getRoutesByType(isMobile).map(route => (
                            <Route
                                key={route.key}
                                path={route.path}
                                exact={route.exact}
                                component={route.sidebar}
                            />
                        ))}
                    </div>
                    <div className="col-xs-9">
                        {routes.getRoutesByType(isMobile).map(route => (
                            <Route
                                key={route.key}
                                path={route.path}
                                exact={route.exact}
                                component={route.main}
                            />
                        ))}
                    </div>
                </div>
            </Fragment>
        );
    }}
    />
);

const Mobile = () => {
    return (
        <Device render={() => {
            return (
                <div>
                    educatopm mpbile
                    {routes.getRoutesByType(true).map(route => (
                        <Route
                            key={route.key}
                            path={route.path}
                            exact={route.exact}
                            component={route.component}
                        />
                    ))}
                </div>
            );
        }}
        />
    );
};

const Dekstop = () => {
    return (
        <Device render={() => {
            return (
                <div>
                    educatopm desktop
                    {routes.getRoutesByType(false).map(route => (
                        <Route
                            key={route.key}
                            path={route.path}
                            exact={route.exact}
                            component={route.component}
                        />
                    ))}
                </div>
            );
        }}
        />
    );
};
// Education.propTypes = {
//     match: PropTypes.shape({}).isRequired,
//     location: PropTypes.shape({}).isRequired
// };

export default [Mobile, Dekstop];
