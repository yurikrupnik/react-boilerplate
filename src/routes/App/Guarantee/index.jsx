import React from 'react';
import { Switch } from 'react-router-dom';
import styles from './styles.scss';
import routes from './routes';
import { Consumer as DeviceConsumer } from '../../services/context/device/index';
import Route from '../../../components/Route';

const Guarantee = () => {
    return (
        <DeviceConsumer render={(renderProps) => {
            const isMobile = renderProps.isMobile();
            if (isMobile) {
                return (
                    <div className={styles.root}>
                        <div className={styles.guaranteeContainer}>
                            <Switch>
                                {routes
                                    .map(route => (
                                        <Route
                                            key={route.key}
                                            path={route.path}
                                            exact={route.exact}
                                            component={route.sidebar}
                                        />
                                    ))}
                            </Switch>
                            <Switch>
                                {routes
                                    .map(route => (
                                        <Route
                                            key={route.key}
                                            path={route.path}
                                            exact={route.exact}
                                            component={route.component}
                                        />
                                    ))}
                            </Switch>
                        </div>
                    </div>
                );
            }
            return (
                <div className={styles.guaranteeContainer}>
                    <h2 className={styles.guaranteeHeader}>
                        The James Allen Guarantee
                    </h2>
                    <div className="row">
                        <div className="col-xs-3">
                            <Switch>
                                {routes
                                    .map(route => (
                                        <Route
                                            key={route.key}
                                            path={route.path}
                                            exact={route.exact}
                                            component={route.sidebar}
                                        />
                                    ))}
                            </Switch>
                        </div>
                        <div className="col-xs-8">
                            <Switch>
                                {routes
                                    .map(route => (
                                        <Route
                                            key={route.key}
                                            path={route.path}
                                            exact={route.exact}
                                            component={route.component}
                                        />
                                    ))}
                            </Switch>
                        </div>
                    </div>
                </div>
            );
        }}
        />
    );
};

export default Guarantee;
