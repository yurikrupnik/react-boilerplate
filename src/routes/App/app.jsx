import React, { Fragment } from 'react';
import { Route } from 'react-router-dom';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core';
import Button from '@material-ui/core/Button';
import Providers from './providers';
import { Provider as ThemeProvider } from '../../components/contexts/themes';
import { Provider as DeviceProvider, Consumer as DeviceConsumer } from './services/device';
import { Provider as SidebarProvider, Consumer as SidebarConsumer } from './sidebar';
import apiProviders from '../../api/providers';
import routes from './routes';
import themeConfig from '../../theme';

const theme = createMuiTheme({
    // palette: {
    //     secondary: brown
    // },
    typography: {
        // Tell Material-UI what the font-size on the html element is.
        // htmlFontSize: 10,
        button: {
            padding: '20px',
            '&:hover': {
                backgroundColor: 'transparent'
            }
            // root: {
            // primary: green,
            // color: {
            //     main: green
            // },
            // root: {
            //     color: blue
            // }
            // }
        }
    },
    palette: {
        primary: {
            main: themeConfig.colors.primary,
        }
    },
    // button: {
    //     root: {
    //         color: blue,
    //     }
    // },
    overrides: {
        // MuiButton: {
        //     // textPrimary: 'red',
        //     // backgroundColor: 'transparent',
        //     disableRipple: false,
        //     root: {
        //         // backgroundColor: 'transparent',
        //         padding: '2px', // works
        //         // color: 'red',
        //         // primary: 'purple',
        //         // color: red,
        //         // padding: '2px',
        //         // '&:hover': {
        //         //     background: 'blue'
        //         // },
        //         '&:hover': {
        //             backgroundColor: 'transparent',
        //         },
        //     },
        // },
        // MuiButtonBase: {
        //     root: {
        //         padding: '2px',
        //     },
        //     // '&:hover': {
        //     //     background: 'blue'
        //     // }
        //     // secondary: blue,
        //     // primary: red,
        //     // color: {
        //     // primary: blue
        //     // },
        //     // The properties to apply
        //     // disableRipple: true, // No more ripple, on the whole application ðŸ’£!
        // },
    }
});

class App extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            appData: (global.window && global.window.appData) || {}
        };
        if (global.window) {
            delete global.window.appData;
        }
    }

    render() {
        // console.log('this.state.appData', this.state.appData);
        const { userAgent } = this.props;
        return (
            <DeviceProvider userAgent={userAgent}>
                <Providers providers={apiProviders.concat(ThemeProvider)}>
                    <MuiThemeProvider theme={theme}>
                        <DeviceConsumer render={(deviceProps) => {
                            const isMobile = deviceProps.isMobile();
                            const basicRoutes = (
                                <Fragment>
                                    <Button onClick={deviceProps.toggle}>Toggle</Button>
                                    {routes.getRoutesByType(isMobile)
                                        .map(route => <Route key={route.key} {...route} />)}
                                </Fragment>
                            );

                            return isMobile ? (
                                <SidebarProvider>
                                    <Fragment>
                                        <SidebarConsumer links={routes.getLinks()} />
                                        {basicRoutes}
                                    </Fragment>
                                </SidebarProvider>
                            ) : basicRoutes;
                        }}
                        />
                    </MuiThemeProvider>
                </Providers>
            </DeviceProvider>
        );
    }
}


// const App = () => (
//     <Providers providers={apiProviders.concat(ThemeProvider, DeviceProvider)}>
//         <MuiThemeProvider theme={theme}>
//             <Device render={(deviceProps) => {
//                 const isMobile = deviceProps.isMobile();
//                 const basicRoutes = (
//                     <Fragment>
//                         {routes.getRoutesByType(isMobile)
//                             .map(route => <Route key={route.key} {...route} />)}
//                     </Fragment>
//                 );
//
//                 return isMobile ? (
//                     <SidebarProvider>
//                         <Fragment>
//                             <SidebarConsumer links={routes.getLinks()} />
//                             {basicRoutes}
//                         </Fragment>
//                     </SidebarProvider>
//                 ) : basicRoutes;
//             }}
//             />
//         </MuiThemeProvider>
//     </Providers>
// );


export default App;
