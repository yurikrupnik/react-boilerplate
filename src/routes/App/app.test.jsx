
import React from 'react';
import { StaticRouter } from 'react-router-dom';
import App from './app';

const {
    it,
    expect,
    mount
} = global;

const desk = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36'
// const mobile = 'Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Mobile Safari/537.36'

it('renders <App /> component', () => {
    const wrapper = mount((
        <StaticRouter context={{}}>
            <App userAgent={desk} />
        </StaticRouter>
    ));
    expect(wrapper).toMatchSnapshot();
});

// it('renders <App /> component', () => {
//     const wrapper = mount((
//         <StaticRouter context={{}}>
//             <App userAgent={mobile} />
//         </StaticRouter>
//     ));
//     expect(wrapper).toMatchSnapshot();
// });
