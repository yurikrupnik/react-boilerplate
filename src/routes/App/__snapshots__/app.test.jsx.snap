// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <App /> component 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App
      staticContext={Object {}}
      userAgent="Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36"
    >
      <Route
        render={[Function]}
      >
        <DeviceProvider
          theme={
            Object {
              "breakpoints": Object {
                "desktop": "1024px",
                "phone": "640px",
                "tablet": "768px",
              },
              "button": Object {
                "background": "linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)",
                "border": 0,
                "borderRadius": 3,
                "boxShadow": "0 3px 5px 2px rgba(255, 105, 135, .3)",
                "color": "white",
                "height": 48,
                "padding": "0 30px",
                "primary": "blue",
              },
              "colors": Object {
                "danger": "#e94b35",
                "desktop": "#303030",
                "error": "#e94b35",
                "mobile": "#656565",
                "primary": "#ff8080",
                "secondary": "#7F8FA4",
                "success": "#1fce6d",
                "warning": "#f2c500",
              },
              "typography": Object {
                "families": Object {
                  "primary": "'Nunito Sans', sans-serif",
                  "secondary": "'Playfair Display', serif",
                },
                "font": "'Nunito Sans', sans-serif",
                "fontSize": "13px",
                "text": "1rem",
                "title": "2rem",
              },
            }
          }
          userAgent="Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36"
        >
          <Providers
            providers={
              Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ]
            }
          >
            <UsersProvider
              initData={Array []}
            >
              <ProjectsProvider>
                <CurrencyProvider
                  initData={Array []}
                >
                  <ThemesProvider>
                    <LoadableComponent>
                      <Loading
                        error={null}
                        isLoading={true}
                        pastDelay={false}
                        retry={[Function]}
                        timedOut={false}
                      />
                    </LoadableComponent>
                  </ThemesProvider>
                </CurrencyProvider>
              </ProjectsProvider>
            </UsersProvider>
          </Providers>
        </DeviceProvider>
      </Route>
    </App>
  </Router>
</StaticRouter>
`;
